search/includes/classes/class-search.php

```diff
@@ -1018,6 +1029,71 @@ public function get_current_host() {
 		return VIP_ELASTICSEARCH_ENDPOINTS[ $this->current_host_index ];
 	}
 
+	/**
+	 * Filter for reducing post meta for indexing to only the allow list and for
+	 * respecting the maximum field count gracefully
+	 */
+	public function filter__ep_prepare_meta_data( $current_meta, $post ) {
+		if ( defined( 'FILES_CLIENT_SITE_ID' ) ) {
+			if ( in_array( FILES_CLIENT_SITE_ID, self::DISABLE_POST_META_ALLOW_LIST, true ) ) {
+				return $current_meta;
+			}
+		}
+
+		if ( ! is_array( $current_meta ) ) {
+			return $current_meta;
+		}
+
+		if ( \is_wp_error( $post ) || ! is_object( $post ) ) {
+			return $current_meta;
+		}
+		
+		/**
+		 * Filters the allow list used for post meta indexing
+		 * 
+		 * @hook vip_search_post_meta_allow_list
+		 * @param {array} $current_allow_list The current allow list for post meta indexing
+		 * @return {array} $new_allow_list The new allow list for post_meta_indexing
+		 */
+		$client_post_meta_allow_list = apply_filters( 'vip_search_post_meta_allow_list', self::POST_META_DEFAULT_ALLOW_LIST );
+
+		// Only include meta that matches the allow list
+		$new_meta = array_intersect( $current_meta, $client_post_meta_allow_list );
+
+		$taxonomies = \get_taxonomies( '', 'names' );
+		$terms = \wp_get_object_terms( $post->ID, $taxonomies );
```

Comment:
If we want to keep a running total like this, we should move these two queries out and into `init` or similar, otherwise they'll run for every post being indexed (which is a ton on a bulk index).